webui/react/src/pages/ExperimentDetails.tsx

```diff
@@ -144,10 +144,23 @@ const ExperimentDetails: React.FC = () => {
   }, [ canceler ]);
 
   useEffect(() => {
-    if (!isSingleTrial || firstTrialId != null) return;
+    if (isSingleTrial === false || firstTrialId != null) return;
     return () => stopPollingFirstTrialId();
   }, [ firstTrialId, isSingleTrial, stopPollingFirstTrialId ]);
 
+  const ExperimentTabs = useMemo(() => {
+    if (!experiment) return null;
+    if (isSingleTrial) {
+      return <ExperimentSingleTrialTabs
+        continueTrialRef={continueTrialRef}
+        experiment={experiment}
+        trialId={firstTrialId}
+      />;
+    } else if (isSingleTrial === false) {
+      return <ExperimentMultiTrialTabs experiment={experiment} />;
+    } else return null;
+  }, [ experiment, firstTrialId, isSingleTrial ]);
+
```

Comment:
**suggestion:** A simpler change might be to change this line:
```
  } else if (!experiment) {
    return <Spinner tip={`Loading experiment ${experimentId} details...`} />;
  }
```
to
```
  } else if (!experiment || isSingleTrial === undefined) {
    return <Spinner tip={`Loading experiment ${experimentId} details...`} />;
  }
```
Then you can avoid having to modify any of the JSX below because it won't reach that return block and by then `isSingleTrial` will be defined as true or false. Because if it's still `undefined` it will return a spinner.